docs/recipes/UsageWithTypescript.md

```diff
@@ -6,259 +6,295 @@ hide_title: true
+    // highlight-start
+    // Use the PayloadAction type to declare the contents of `action.payload`
+    incrementByAmount: (state, action: PayloadAction<number>) => {
+      // highlight-end
+      state.value += action.payload
     }
   }
-}
+})
+
+export const { increment, decrement, incrementByAmount } = counterSlice.actions
+
+// Other code such as selectors can use the imported `RootState` type
+export const selectCount = (state: RootState) => state.counter.value
+
+export default counterSlice.reducer
 ```
 
-System Action Constants & Shape:
+The generated action creators will be correctly typed to accept a `payload` argument based on the `PayloadAction<T>` type you provided for the reducer. For example, `incrementByAmount` requires a `number` as its argument.
 
-```ts
-// src/store/system/types.ts
-export const UPDATE_SESSION = 'UPDATE_SESSION'
+### Use Typed Hooks in Components
 
-interface UpdateSessionAction {
-  type: typeof UPDATE_SESSION
-  payload: SystemState
-}
+In component files, import the pre-typed hooks instead of the standard hooks from React Redux.
 
-export type SystemActionTypes = UpdateSessionAction
-```
+```tsx title="features/counter/Counter.tsx"
+import React, { useState } from 'react'
 
-With these types we can now also type check system's action creators:
+// highlight-next-line
+import { useAppSelector, useAppDispatch } from 'app/hooks'
 
-```ts
-// src/store/system/actions.ts
+import { decrement, increment } from './counterSlice'
 
-import { SystemState, UPDATE_SESSION, SystemActionTypes } from './types'
+export function Counter() {
+  // highlight-start
+  // The `state` arg is correctly typed as `RootState` already
+  const count = useAppSelector(state => state.counter.value)
+  const dispatch = useAppDispatch()
+  // highlight-end
 
-export function updateSession(newSession: SystemState): SystemActionTypes {
-  return {
-    type: UPDATE_SESSION,
-    payload: newSession
-  }
+  // omit rendering logic
 }
 ```
 
+## Typing Additional Redux Logic
+
 ### Type Checking Reducers
 
-Reducers are just pure functions that take the previous state, an action and then return the next state. In this example, we explicitly declare the type of actions this reducer will receive along with what it should return (the appropriate slice of state). With these additions TypeScript will give rich intellisense on the properties of our actions and state. In addition, we will also get errors when a certain case does not return the `ChatState`.
+[Reducers](../tutorials/fundamentals/part-3-state-actions-reducers.md) are pure functions that receive the current `state` and incoming `action` as arguments, and return a new state.
 
-Type checked chat reducer:
+If you are using Redux Toolkit's `createSlice`, you should rarely need to specifically type a reducer separately. If you do actually write a standalone reducer, it's typically sufficient to declare the type of the `initialState` value, and type the `action` as `AnyAction<string>`:
```

Comment:
`AnyAction` does not accept any arguments, so it's just `AnyAction`, not `AnyAction<string>`